en/codegen/layout-codegen.md

```diff
@@ -25,43 +25,112 @@ In the below image the developer would select the Sketch Groups `peopleList` and
 Normally Angular components are sized or resized by their parent at runtime. However, there are design scenarios where the designer wants the component to be a fixed size element when added to the Angular application. Code generation supports this design scenario.
 
-### After Component Code Generation – Before Rendering in The Browser
-
-Developers need to perform the following edits of generated components before rendering in the browser:
-
-- If the HTML has formControlName elements, the formGroup comment under the `<form>` tag must be uncommented: `<!-- [formGroup]="customerForm" -->`
-- In the TypeScript, check for `<!-- // TODO – uncomment-->` and uncomment the FormGroup comment if you have one.
-- In the TypeScript, check the `ngOnInit` method and uncomment the form code.
-
 ## Sketch Element Names
 
 Code generation requires each element in the Sketch Layer List to have a name. Any element without a name will be skipped and not processed. This includes all of the skipped element's children as well.
 
 ## Layouts
 
-Layouts are rendered with absolute position and sizes. Each component has its own position (left, top) and size (width, height) that determine how it will be rendered on the page.
+Sketch arranges artboards of projects through absolute positioning, using properties such as Top, Left, Width and Height. Often these designs have to represent fluid and responsive applications. Most web applications built from such designs use modern paradigms such as flex and grid display containers. The job of the Code generation is to get as close as possible to the application, a developer would build from a given design. This requires the Code generation to go through a set of heuristics and based on them to produce fluid design from the absolute/static one that was created in Sketch.
+
+Code generation reads Sketch files and uses the layout properties defined by native Sketch schema. Please be aware, if a 3rd party Sketch plugin is used to create and maintain the Sketch drawing layout, they typically have their own set of properties that code generation is not aware of and will not use when creating the HTML and CSS.
+Code generation respects and uses Sketch groups when creating the layout. Code generation renders HTML CSS that uses [Flexbox](https://css-tricks.com/snippets/css/a-guide-to-flexbox/). Flexbox lays out elements in either a column or row.
+Sketch container elements such as a group or an artboard are rendered as divs with flex CSS applied to them.
+
+### How it works
+Main idea of building responsive layout is to combine the components into groups. This creates a structure, thats expected from the developer. From Code generation point of view,and because of the Flexbox, these groups should be rows or columns. And after the layout is created, the components can end up into an hierarchy of rows and columns depending on their positions and Sketch groups.
```

Comment:
~~And after the layout is created, the components can end up into an hierarchy of rows and columns depending on their positions and Sketch groups.~~

After the layout is created components are placed in a hierarchy of rows and columns depending on their positions relative to other components and their respective Sketch groups.